### PR TITLE
CC - ropes issue7

### LINKED ISSUE #7: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [x] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [x] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [x] Where in the code should we implement this functionality?
- [x] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by calvincui233**
Closes #7 

Add tag filters. Clicking a tag will navigate to index page displaying images with that tag.

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -98,4 +98,24 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+  end
+
+  test 'test empty tag' do
+    get images_path, params: { tag: '' }
+    assert_response :ok
+    assert_select 'h1', 'No Images Found'
+  end
+
+  test 'test existent tag' do
+    Image.create!(url: 'https://bit.ly/2siExH7', tag_list: ['Santa Barbara', 'landscape', 'beach'])
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Should we add to this test the condition that there is another `Image` that exists but does not have the desired tag?

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -5,7 +5,11 @@ class ImagesController < ApplicationController
   protect_from_forgery with: :exception
 
   def index
-    @images = Image.order('updated_at DESC')
+    @images = if params[:tag].blank?
+                Image.order('created_at DESC')
+              else
+                Image.order('created_at DESC').tagged_with(params[:tag])
```
[the remark this answers is not in the document]
Comment (calvincui233) [reply app/controllers/images_controller.rb]: 😎

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (calvincui233) [reply test/controllers/images_controller_test.rb]: Thanks for pointing it out! Just added one picture without desired tag.

Comment (calvincui233) [reply test/controllers/images_controller_test.rb]: Thanks for pointing it out! Just added one picture without desired tag.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -98,4 +98,25 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    assert_select 'h1', 'No Images Found'
+  end
+
+  test 'test existent tag' do
+    Image.create!(url: 'https://bit.ly/2CfKXeF')
+    Image.create!(url: 'https://bit.ly/2siExH7', tag_list: ['Santa Barbara', 'landscape', 'beach'])
+
+    get images_path, params: { tag: 'beach' }
+    assert_response :ok
+    assert_select '.image-card', count: 1
```
Comment (almgong) [inline test/controllers/images_controller_test.rb]: Should add another assertion to make sure that the correct image was returned? Maybe we could use the URL?

Comment (calvincui233) [reply test/controllers/images_controller_test.rb]: Image url is now applied to make sure correct image is returned.

**Pull request merged**